File: Algorithms/WeekOfCode/Week25/BetweenTwoSets.java
package WeekOfCode.Week25;

import java.util.Arrays;
import java.util.Scanner;

/**
 *	Algorithms -> Week of Code 25 -> Between Two Sets 
 *	Easy
 *
 * Sample Input:
   2 3
   2 4
   16 32 96
 *
 * Sample Output:
   3
 */
public class BetweenTwoSets {

	public static void main(String[] args) {
		
		Scanner in = new Scanner(System.in);
		
		int n = in.nextInt();
		int m = in.nextInt();
		
		int[] A = new int[n];
		
		int[] B = new int[m];
		
		for(int i = 0; i < n; i++) {
			A[i] = in.nextInt();
		}
		
		for(int i = 0; i < m; i++) {
			B[i] = in.nextInt();
		}
		
		in.close();
		
		Arrays.sort(A);
		Arrays.sort(B);
		
		int gcdB = B[0];
		
		for(int i = 1; i < m; i++) {
			gcdB = GCD(gcdB, B[i]);
		}
		
		//System.out.println("gcd of B: " + gcdB);
		
		int lcmA = A[0];
		
		for(int i = 1; i < n; i++) {
			lcmA = LCM(lcmA, A[i]);
		}
		
		//System.out.println("lcm of A: " + lcmA);
		
		int counter = 0;
		for(int i = lcmA; i <= gcdB; i++) {
			if(isFactor(gcdB, i)) {
				counter++;
				//System.out.println(i);
			}
		}
		
		System.out.println(counter);
	}
	
	// Simple Algorithm to calculate GCD
	public static int GCD(int a, int b) 
	{
		if (b==0) return a;
		return GCD(b,a%b);
	}
	
	// Simple Algorithm to calculate LCM
	public static int LCM(int a, int b) 
	{
		return Math.abs(a * b)/ GCD(a, b);
	}
	
	// Determines if divider is a factor of number
	public static boolean isFactor(int number, int divider)
	{
		return number % divider == 0;
	}
}

File: Algorithms/Implementation/BetweenTwoSets.c
#include <math.h>
#include <stdio.h>
#include <string.h>
#include <stdlib.h>
#include <assert.h>
#include <limits.h>
#include <stdbool.h>

/**
 *	Algorithms -> Warmup -> Between Two Sets
 *	Easy
 */

/*
 * Function declaration
 */
int gcd(int a, int b);
int lcm(int a, int b);
int isFactor(int number, int divisor);

/*
 * Main
 */
int main(){

	int sizeOfA, sizeOfB;

	scanf("%d %d", &sizeOfA, &sizeOfB);

	int A[sizeOfA];

	for(int i = 0; i < sizeOfA; i++) {
		scanf("%d", &A[i]);
	}

	int B[sizeOfB];

	for(int i = 0; i < sizeOfB; i++) {
		scanf("%d", &B[i]);
	}

	int gcdB = B[0];

	for(int i = 1; i < sizeOfB; i++) {
		gcdB = gcd(gcdB, B[i]);
	}

	int lcmA = A[0];

	for(int i = 1; i < sizeOfA; i++) {
		lcmA = lcm(lcmA, A[i]);
	}

	int counter = 0;

	for(int i = lcmA; i <= gcdB; i++) {
		if(isFactor(gcdB, i) == 1 && isFactor(i, lcmA) == 1) {
			counter++;

		}
	}

	printf("%d", counter);

    return 0;
}

/**
 * Functions
 */

// Calculate the greatest common denominator of two numbers.
int gcd(int a, int b)
{
	if(b == 0) return a;
	return gcd(b, a % b);
}

// Calculate the least common multiple of two numbers.
int lcm(int a, int b)
{
	return abs(a * b) / gcd(a, b);
}

// Determines if a number is a divisor of another number
int isFactor(int number, int divisor)
{
	int isDivisor = 0;
	if(number % divisor == 0) {
		isDivisor = 1;
	}
	return isDivisor;
}
